records_mover/db/redshift/loader.py

```diff
@@ -114,19 +112,20 @@ def can_load_this_format(self, source_records_format: BaseRecordsFormat) -> bool
             processing_instructions = ProcessingInstructions()
             load_plan = RecordsLoadPlan(records_format=source_records_format,
                                         processing_instructions=processing_instructions)
-            if not isinstance(load_plan.records_format, DelimitedRecordsFormat):
-                return False
-            unhandled_hints = set(load_plan.records_format.hints.keys())
+            unhandled_hints = set()
+            records_format = load_plan.records_format
+            if isinstance(records_format, DelimitedRecordsFormat):
+                unhandled_hints = set(records_format.hints.keys())
             processing_instructions = load_plan.processing_instructions
-            hints = load_plan.records_format.\
-                validate(fail_if_cant_handle_hint=processing_instructions.fail_if_cant_handle_hint)
             redshift_copy_options(unhandled_hints,
-                                  hints,
+                                  records_format,
                                   processing_instructions.fail_if_cant_handle_hint,
                                   processing_instructions.fail_if_row_invalid,
                                   processing_instructions.max_failure_rows)
-            complain_on_unhandled_hints(processing_instructions.fail_if_dont_understand,
-                                        unhandled_hints, load_plan.records_format.hints)
+            if isinstance(records_format, DelimitedRecordsFormat):
+                complain_on_unhandled_hints(processing_instructions.fail_if_dont_understand,
+                                            unhandled_hints,
+                                            records_format.hints)
```

Comment:
The changes above are to stop assuming we're dealing with a DelimitedRecordsFormat object and let the other types through.